aten/src/ATen/native/mkl/LinearAlgebra.cpp

```diff
@@ -100,6 +133,63 @@ static inline void gemm_batched(const CBLAS_TRANSPOSE trans_A, const CBLAS_TRANS
     reinterpret_cast<const void*>(&beta), reinterpret_cast<void**>(C), &ldc, 1, &batch_size);
 }
 
+static CBLAS_TRANSPOSE to_cblas(TransposeType x) {
+  switch (x) {
+    case TransposeType::NoTranspose: return CblasNoTrans;
+    case TransposeType::Transpose: return CblasTrans;
+    case TransposeType::ConjTranspose: return CblasConjTrans;
+  }
+  TORCH_INTERNAL_ASSERT(false, "Unknown TransposeType");
+}
+
+void mkl_gemm_batched(
```

Comment:
it seems these functions are the same as `gemm_batched` above, modulo type of trans_A/trans_B, why do we need 2 copies?

Reply:
The biggest difference is `ColMajor` vs. `RowMajor`. In a follow up I can remove `_baddbmm_mkl_` entirely though. Just have the normal version call `cpublas::gemm_batched_with_stride`.